### PR TITLE
Make BibEntryPredictor conform to MMDA interface.

### PR DESCRIPTION
**Pull request by cmwilhelm**
+ takes in Document with expected `bib_entry_boxes`
    -> not thrilled about this key name, since it immediately becomes SpanGroups
+ original `predict()` method preserved as `predict_raw()`
+ new TIMO predictor added for mmda variant (old one preserved however)

NOTE: mapping inferred spans back onto the Document is the touchiest
part of this code, as underlying model requires full bib entry strings.
Must account for gaps between spans in source document SpanGroup when
recovering offsets and projecting into the doc.

### CODE REVIEW

File: mmda/predictors/hf_predictors/bibentry_predictor/utils.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        # Build up relative positions for each SG span within the SG text
+        intervals = []
+        curr_total = 0
+        for span in sg.spans:
+            i_start = curr_total
+            curr_total += span.end - span.start
+            intervals.append((i_start, curr_total))
+            curr_total += len(_SPAN_JOINER)
+
+        def map_raw_span(target, raw_span):
```
Comment (geli-gel) [inline mmda/predictors/hf_predictors/bibentry_predictor/utils.py:45]: 😵‍💫

**Review by geli-gel — APPROVED** [review]
I think it looks good! 👍🏼 
that mapping part 😵‍💫

File: mmda/predictors/hf_predictors/bibentry_predictor/utils.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                        # Continues into next span, or ends at exact end of this one.
+                        new_sg_span_end = sg_span.end
+                    else:
+                        # Ends in the middle of this span.
+                        new_sg_span_end = new_sg_span_start + end - cur_pos
+
+                    new_spans.append(Span(start=new_sg_span_start, end=new_sg_span_end))
+
+                    # Advance our current position, accounting for whitespace before beginning of next span.
+                    cur_pos = interval_end + 1
```
[the remark this answers is not in the document]
Comment (cmwilhelm) [reply mmda/predictors/hf_predictors/bibentry_predictor/utils.py:70]: yes, you're right!

**Review by kyleclo — APPROVED** [review]
lgtm; offline comments have been addressed

**Pull request merged**